File: kitten-house-care/src/main/java/org/jesperancinha/housing/repository/CatRepositoryImpl.java
package org.jesperancinha.housing.repository;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.jesperancinha.housing.model.Cat;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

import java.io.IOException;

@Repository
public class CatRepositoryImpl {

    private final Cat cat1;
    private final Cat cat2;

    public CatRepositoryImpl(ObjectMapper objectMapper) throws IOException {
        this.cat1 = objectMapper.readValue(getClass().getResourceAsStream("/cat1.json"), Cat.class);
        this.cat2 = objectMapper.readValue(getClass().getResourceAsStream("/cat2.json"), Cat.class);
    }

    public Mono<Cat> getCatById(Long id) {
        return Mono.fromCallable(() -> getCat(id));
    }

    public Cat getCatByIdNonReactive(Long id) {
        return getCat(id);
    }

    private Cat getCat(Long id) {
        if (id.intValue() == 1L) {
            return cat1;
        }
        if (id.intValue() == 2L) {
            return cat2;
        }
        return null;
    }
}

File: kitten-house-care/src/main/kotlin/org/jesperancinha/housing/repository/CatRepositoryImpl.kt
package org.jesperancinha.housing.repository

import com.fasterxml.jackson.databind.ObjectMapper
import org.jesperancinha.housing.model.Cat
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
class CatRepositoryImpl(objectMapper: ObjectMapper) {
    private val cat1: Cat
    private val cat2: Cat

    init {
        cat1 = objectMapper.readValue(javaClass.getResourceAsStream("/cat1.json"), Cat::class.java)
        cat2 = objectMapper.readValue(javaClass.getResourceAsStream("/cat2.json"), Cat::class.java)
    }

    fun getCatById(id: Long): Mono<Cat?> {
        return Mono.fromCallable { getCat(id) }
    }

    fun getCatByIdNonReactive(id: Long): Cat? {
        return getCat(id)
    }

    private fun getCat(id: Long): Cat? {
        if (id.toInt() == 1L) {
            return cat1
        }
        return if (id.toInt() == 2L) {
            cat2
        } else null
    }
}